### PR TITLE
Update for project structure build and packaging stability

### PR DESCRIPTION
**Pull request by brismithers**
# Description
## One Line Summary
Increase project structure build and deploy stability.

### Project Structure
* The `OneSignalSDK.DotNet` project is now a multi-platform project targeting `net6.0`, `net6.0-android`, and `net6.0-ios`. It now depends on the `OneSignalSDK.DotNet.Core/iOS/Android` projects and is the "main entry" to the SDK.  `OneSignalSDK.DotNet.iOS/Android` projects, which were the "main entry" to the SDK, are now a more standard dependent library and produce their own `OneSignal.DoeNet.iOS/Android.dll` files.  This was done in place of having the `OneSignalSDK.DotNet.iOS/Android` projects compile link to the source files within `OneSignalSDK.DotNet` project where both projects built a platform-specific `OneSignalSDK.DotNet.dll`, which can confuse the build system.
* Fix the `OneSignalSDK.DotNet.Android` project to properly use the new `PackageReference` format, remove `packages.config` file and associated `Imports`/`References` within the `OneSignalSDK.DotNet.Android.csproj` file.  The old `packages.config` system was not being used, and this was confusing the build system.
* Create unique namespaces for the `OneSignalSDK.DotNet.iOS/Android` projects and keep the namespaces in line with the folder structure.
* Remove explicit `ReleaseVersion` from all `.csproj` files.  The expectation is the version will be supplied at build time (i.e. `dotnet build -c Release OneSignal.sln /p:Version=4.2.1`).

### Packaging
* File includes within the `nuspec` file updated to reflect the new project structure and build output.
* File includes within the `nuspec` file are now explicit.  When specifying wildcards there is no explicit checking to ensure all required files have been included in the package.  Now that files are explicitly specified, if the file does not exist the package will not be created.
* The `license` property within the `nuspec` file is now used instead of the deprecated `licenseUrl`.

### Sample Project
* Update sample project to reference SDK via project rather than nuget package, to allow for quicker change/build/test cycles.

### ⚠️  BREAKING CHANGE ⚠️ 
The fullname for `OneSignalSDK.DotNet.OneSignalImplementation` in the net6.0-ios target has been updated to `OneSignalSDK.DotNet.iOS.OneSignalImplementation`.  This change may require a coding change within your user notification service extension to include the new naming when calling `OneSignalSDK.DotNet.iOS.OneSignalImplementation.DidReceiveNotificationExtensionRequest`.

### Motivation
The desire to have the sample project reference the SDK as a project reference rather than as a nuget package is preferred as it allows changes to be made within the SDK and immediately tested via the sample project.  It is also the case, in order to build the SDK reliably a manual sequence of project builds would have to be done to ensure a successful build.

In order for the sample project to reference the SDK via project reference and produce a more reliable build, a more conventional project dependency pattern needed to be established.

### Scope
No functional changes are part of this change, only build/packaging related changes.  The "net change" is the resulting `nupkg` has a `OneSignalSDK.DotNet.dll` which no longer houses class types from the `OneSignalSDK.DotNet.iOS/Android` projects.  There is now a `OneSignalSDK.DotNet.Android.dll` in the `net6.0-android21.0` target, and an additional `OneSignalSDK.DotNet.iOS.dll` in the `net6.0-ios10.0` which houses the respective class types.

This build/packaging change should be transparent to a consumer of the nuget package, as the `nupkg` has always included multiple DLLS required for the SDK.  Note this is also the reason why we must still rely on a `nuspec` file, rather than using the `csproj` for `dotnet pack`.

# Testing

## Manual testing
* Built the sample project (now referencing via project reference) and ensure correct build/link/deploy.
* Created a standalone MAUI app which referenced the SDK via local `nupkg`, the `nupkg` built using the new structure and ensured correct compilation/deployment.  At runtime confirmed device registration and that a notification could be received by the device.  The main focus was to ensure the Android/iOS native SDKs were successfully packaged/linked within the `nupkg`.

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [ ] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

### CODE REVIEW

File: Samples/OneSignalApp/exampledotnet/exampledotnet.csproj [hunk 1]
```diff
@@ -27,6 +27,12 @@
 		<ReleaseVersion>4.2.0-beta1-maui-only</ReleaseVersion>
```
Comment (jkasten2) [inline Samples/OneSignalApp/exampledotnet/exampledotnet.csproj:27]: Missed cleaning up this ReleaseVersion entry.

File: issue_template.md [hunk 2]
```diff
@@ -1,42 +0,0 @@
-<!--
```
Comment (jkasten2) [inline issue_template.md:1]: Looks like this issue template was deleted by mistake.

File: Samples/OneSignalNotificationServiceExtension/NotificationService.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using Foundation;
 using OneSignalSDK.DotNet;
+using OneSignalSDK.DotNet.iOS;
```
Comment (jkasten2) [inline Samples/OneSignalNotificationServiceExtension/NotificationService.cs:4]: Looks like this is a breaking change, however there isn't many integrations yet, and even fewer who probably have setup the iOS NSE. However let's include a note of this breaking change in the PR as well as the release notes.

File: OneSignalSDK.dotnet/OneSignalSDK.DotNet.cs [hunk 4]
```diff
@@ -21,15 +21,18 @@ public static OneSignalSDKInternal Default
...
-         #else
-            return new OneSignalImplementation();
-         #endif
-      }
+#endif
+
+        }
 
-      internal static Exception NotImplementedInReferenceAssembly()
+        internal static Exception NotImplementedInReferenceAssembly()
```
Comment (jkasten2) [inline OneSignalSDK.dotnet/OneSignalSDK.DotNet.cs:35]: nit, inconsistent whitespace

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: issue_template.md (hunk 2 above)
Comment (brismithers) [reply issue_template.md:1]: I don't think this issue template is being used.  The issue templates being used are [here](https://github.com/OneSignal/OneSignal-DotNet-SDK/tree/main/.github/ISSUE_TEMPLATE)

File: Samples/OneSignalApp/exampledotnet/exampledotnet.csproj (hunk 1 above)
Comment (brismithers) [reply Samples/OneSignalApp/exampledotnet/exampledotnet.csproj:27]: Missed it here and in the sample OneSignalNotificationServiceExtension project, removed from both

File: Samples/OneSignalNotificationServiceExtension/NotificationService.cs (hunk 3 above)
Comment (brismithers) [reply Samples/OneSignalNotificationServiceExtension/NotificationService.cs:4]: agree, and agree

File: OneSignalSDK.dotnet/OneSignalSDK.DotNet.cs (hunk 4 above)
Comment (brismithers) [reply OneSignalSDK.dotnet/OneSignalSDK.DotNet.cs:35]: fixed

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**